forest_lib/forest/plot.py

```diff
@@ -1011,110 +635,63 @@ def update_colorbar_widget(self):
         except AttributeError as e1:
             print('Unable to update colorbar as colorbar widget not initiated')
 
-    @forest.util.timer
     def set_data_time(self, new_time):
-
-        '''
-
-        '''
-
         print('selected new time {0}'.format(new_time))
-
         self.current_time = new_time
-        # self.update_plot()
-        self.create_matplotlib_fig()
-        if not self.async:
-            self.update_bokeh_img_plot_from_fig()
-            if self.stats_widget:
-                self.update_stats_widget()
+        if self.visible:
+            self.render()
             if self.colorbar_widget:
                 self.update_colorbar_widget()
 
     def set_var(self, new_var):
-
-        '''
-
-        '''
-
         print('selected new var {0}'.format(new_var))
-
         self.current_var = new_var
-        self.create_matplotlib_fig()
-        if not self.async:
-            self.update_bokeh_img_plot_from_fig()
-            if self.stats_widget:
-                self.update_stats_widget()
+        if self.visible:
+            self.render()
             if self.colorbar_widget:
                 self.update_colorbar_widget()
 
-    def set_region(self, new_region):
-
-        '''Event handler for a change in the selected plot region.
-
-        '''
-
-        print('selected new region {0}'.format(new_region))
-
-        self.current_region = new_region
-        self.data_bounds = self.region_dict[self.current_region]
-        self.create_matplotlib_fig()
-        if not self.async:
-            self.update_bokeh_img_plot_from_fig()
-            if self.stats_widget:
-                self.update_stats_widget()
+    def set_region(self, region):
+        """Adjust bokeh figure extents"""
+        self.current_region = region
+        extents = self.region_dict[self.current_region]
+        y_start, y_end, x_start, x_end = extents
+        self.bokeh_figure.x_range.start = x_start
+        self.bokeh_figure.x_range.end = x_end
+        self.bokeh_figure.y_range.start = y_start
+        self.bokeh_figure.y_range.end = y_end
```

Comment:
do we need `self.render()` I can see we might not as we are just in bokeh axis land, but the other funcs in this area do.

Reply:
I'm torn between `self.render()` plotting a sub-set of the data or offloading the high-resolution responsibility to another `View` that is watching the axes

The other thing is, should `ForestPlot` control the `bokeh_figure.x_range` at all? As in should we pull the region responsibility out of `ForestPlot`? i.e. remove `region_dict` and `current_region` constructor parameters